src/main/java/org/prebid/server/bidder/readpeak/ReadPeakBidder.java

```diff
@@ -0,0 +1,204 @@
+        final List<Imp> modifiedImps = new ArrayList<>();
+        final List<ExtImpReadPeak> extImpReadPeaks = new ArrayList<>();
+        final List<BidderError> errors = new ArrayList<>();
+
+        for (Imp imp : request.getImp()) {
+            try {
+                final ExtImpReadPeak extImpReadPeak = parseImpExt(imp);
+                final Imp modifiedImp = modifyImp(imp, extImpReadPeak);
+                modifiedImps.add(modifiedImp);
+                extImpReadPeaks.add(extImpReadPeak);
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
+
+        if (modifiedImps.isEmpty()) {
+            return Result.withError(BidderError
+                    .badInput(String.format("Failed to find compatible impressions for request %s", request.getId())));
+        }
+
+        final BidRequest modifiedRequest = request.toBuilder().imp(modifiedImps).build();
+        final HttpRequest<BidRequest> httpRequest = makeHttpRequest(modifiedRequest, extImpReadPeaks.get(0));
+
+        return Result.of(Collections.singletonList(httpRequest), errors);
+    }
+
+    private ExtImpReadPeak parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), READPEAK_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException("Failed to deserialize ReadPeak extension: " + e.getMessage());
+        }
+    }
+
+    private Imp modifyImp(Imp imp, ExtImpReadPeak extImpReadPeak) {
+        return imp.toBuilder()
+                .bidfloor(extImpReadPeak.getBidFloor() != null ? extImpReadPeak.getBidFloor() : imp.getBidfloor())
+                .tagid(StringUtils.isNotBlank(extImpReadPeak.getTagId()) ? extImpReadPeak.getTagId() : null)
+                .build();
+    }
+
+    private HttpRequest<BidRequest> makeHttpRequest(BidRequest request, ExtImpReadPeak extImpReadPeak) {
+        final Publisher publisher = Publisher.builder()
+                .id(extImpReadPeak.getPublisherId())
+                .build();
+
+        final BidRequest.BidRequestBuilder requestBuilder = request.toBuilder();
+
+        if (request.getSite() != null) {
+            final Site site = request.getSite().toBuilder()
+                    .id(StringUtils.isNotBlank(extImpReadPeak.getSiteId())
+                            ? extImpReadPeak.getSiteId() : request.getSite().getId())
+                    .publisher(publisher)
+                    .build();
+            requestBuilder.site(site);
+        } else if (request.getApp() != null) {
+            final App app = request.getApp().toBuilder()
+                    .id(StringUtils.isNotBlank(extImpReadPeak.getSiteId())
+                            ? extImpReadPeak.getSiteId() : request.getApp().getId())
+                    .publisher(publisher)
+                    .build();
+            requestBuilder.app(app);
+        }
```

Comment:
nitpick: I'd create separate methods `modifySite` and `modifyApp`

and then the code could be like the following:
```
boolean hasSite = request.getSite() != null;
boolean hasApp = !hasSite && request.getApp() != null;
final BidRequest outgoingRequest = request.toBuilder()
     .site(hasSite ? modifySite(...) : request.getSite())
     .app(hasApp ? modifyApp(...) : request.getApp())
     .build();
```